nixos/modules/services/video/epgstation/default.nix

```diff
@@ -0,0 +1,297 @@
+      else "INSECURE_GO_CHECK_CONFIGURATION_NIX";
+  in
+  pkgs.writeScript "epgstation-prestart" ''
+    #!${pkgs.runtimeShell}
+
+    PASSWORD="${readPassword cfg.basicAuth.passwordFile}"
+    DB_PASSWORD="${readPassword cfg.database.passwordFile}"
+
+    # setup configuration
+    sed \
+      -e "s,@password@,$PASSWORD,g" \
+      -e "s,@dbPassword@,$DB_PASSWORD,g" \
+      ${settingsTemplate} > /etc/epgstation/config.json
+    chmod 440 /etc/epgstation/config.json
+    chown "${username}:${groupname}" /etc/epgstation/config.json
+
+    # XXX: Use password authentication, since mysqljs does not yet support auth_socket
+    if [ ! -e /var/lib/epgstation/db-created ]; then
+      ${pkgs.mysql}/bin/mysql -e \
+        "GRANT ALL ON ${cfg.database.name}.* TO ${username}@localhost IDENTIFIED by \"$DB_PASSWORD\";"
+      touch /var/lib/epgstation/db-created
+    fi
+  '';
+
+  logConfig = {
+    appenders.stdout.type = "stdout";
+    categories = {
+      default = { appenders = [ "stdout" ]; level = "info"; };
+      system = { appenders = [ "stdout" ]; level = "info"; };
+      access = { appenders = [ "stdout" ]; level = "info"; };
+      stream = { appenders = [ "stdout" ]; level = "info"; };
+    };
+  };
+in
+{
+  options.services.epgstation = {
+    enable = mkEnableOption pkgs.epgstation.meta.description;
+
+    usePreconfiguredStreaming = mkOption {
+      type = types.bool;
+      default = true;
+      description = "Use preconfigured streaming options";
+    };
+
+    readOnlyOnce = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Don't reload configuration files at runtime.";
+    };
+
+    serverPort = mkOption {
+      type = types.port;
+      default = 20772;
+      description = ''
+        Port for EPGStation to listen on. Be careful because EPGStation uses
+        a total of two ports: the one specified by this option for the web
+        interface, and the one above for websockets.
+      '';
+    };
+
+    allowedIPs = mkOption {
+      type = with types; nullOr (listOf str);
+      default = null;
+      example = [ "127.0.0.1" "192.168.1.1/24" ];
+      description = "List of IP addresses allowed to connect to EPGStation";
+    };
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Open ports in the firewall for the EPGStation web interface.
+      '';
+    };
+
+    basicAuth = {
+      user = mkOption {
+        type = with types; nullOr str;
+        default = null;
+        example = "epgstation";
+        description = ''
+          Basic auth username for EPGStation. If null, basic auth will be
+          disabled.
+        '';
+      };
+
+      password = mkOption {
+        type = with types; nullOr str;
+        default = null;
+        example = "password";
+        description = ''
+          Basic auth password for EPGStation.
+
+          Warning: This is implemented to store the plain text in the nix store.
```

Comment:
Should mention the `passwordFile` alternative and that the nix store is world-readable. Also, is this option actually used? It seems like the init script only uses `passwordFile`.

Reply:
The default option for `passwordFile` uses `password`.

Reply:
Fixed description.